js/zb.js

```diff
@@ -995,6 +996,15 @@ module.exports = class zb extends Exchange {
         return await this.spotV1PrivateGetCancelOrder (this.extend (request, params));
     }
 
+    async cancelAllOrders (symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'symbol': market['id'],
+        };
+        return await this.contractV2PrivatePostTradeCancelAllOrders (this.extend (request, params));
```

Comment:
Let's check if they support canceling all orders on spot against their api doc, we may be missing some endpoints they might have added. If they don't support canceling all orders on spot markets, then we should throw a `NotSupported` exception from this method when the user specifies a market of spot type.
Also, if this exchange requires the market id, then we must throw `ArgumentsRequired` if the symbol is not defined, cause it has a default value of `undefined` and the user might have not specified it, so a subsequent call to `this.market()` will fail here.